model/change_point_detection.go

```diff
@@ -428,3 +433,44 @@ func createChangePoint(ctx context.Context, env cedar.Environment, resultToUpdat
 	_, err := env.GetDB().Collection(perfResultCollection).UpdateOne(ctx, filter, update)
 	return errors.Wrap(err, "Unable to create change point")
 }
+
+func TriageChangePoints(ctx context.Context, env cedar.Environment, changePoints map[string]string, status TriageStatus) error {
+	session, err := env.GetDB().Client().StartSession()
+	if err != nil {
+		return errors.Wrap(err, "Unable to start session for triage update")
+	}
+	defer session.EndSession(ctx)
+	if err := session.StartTransaction(); err != nil {
+		return errors.Wrap(err, "Unable to start transaction for triage update")
+	}
+	return mongo.WithSession(ctx, session, func(sc mongo.SessionContext) error {
+		for perfResultId, measurement := range changePoints {
+			filter := bson.M{
+				perfIDKey: perfResultId,
+				bsonutil.GetDottedKeyName(perfAnalysisKey, perfAnalysisChangePointsKey, perfChangePointMeasurementKey): measurement,
+			}
+			update := bson.M{
+				"$set": bson.M{
+					bsonutil.GetDottedKeyName(perfAnalysisKey, perfAnalysisChangePointsKey, "$", perfChangePointTriageKey, perfTriageInfoStatusKey):    status,
+					bsonutil.GetDottedKeyName(perfAnalysisKey, perfAnalysisChangePointsKey, "$", perfChangePointTriageKey, perfTriageInfoTriagedOnKey): time.Now(),
+				},
+			}
+			res, err := env.GetDB().Collection(perfResultCollection).UpdateOne(sc, filter, update)
+			if err != nil {
+				err2 := session.AbortTransaction(ctx)
+				if err2 != nil {
+					return errors.Wrap(err, "Failed to abort transaction during failed change point triage")
+				}
+				return errors.Errorf("Unable to triage change point <%s> for performance result %s", measurement, perfResultId)
+			}
+			if res.ModifiedCount != 1 {
+				return errors.Errorf("Could not find change point <%s> for performance result %s", measurement, perfResultId)
```

Comment:
You probably want to abort here as well

Reply:
Nice catch. FMI: if we don't commit before returning, transaction will abort on its own as a safeguard, right?